### DISCUSSION TITLE
How to Implement authentication in a hexagonal/DDD architecture

### DISCUSSION THREAD

**Discussion by bernard-ng** [symfony/symfony#45863] [Q&A] [answered]
Hi everyone, I'm learning DDD using symfony, and I'm implementing the authentication

I have a **User** entity which is in the auth domain, but I realize that I have to implement the **UserInterface** and **PasswordAuthenticatedUserInterface** of the symfony security component for my entity to be considered as a "user".

Is there any other way to avoid using symfony interfaces in my domain?

**Comment by darthf1** [comment, marked as answer]
You can create a SecurityUser class in your Infrastructure folder, implement those interfaces there, and pass your User entity in the constructor. Methods in the Interface can then be forwarded to your entity. 

```php
public function getUserIdentifier(): string
{
  return $this->entity->getEmail();
}
```

- https://stovepipe.systems/post/decoupling-your-security-user
- https://simshaun.medium.com/decoupling-your-application-user-from-symfonys-security-user-60fa31b4f7f2

**Reply by bernard-ng** [reply]
Thanks a lot
